### DISCUSSION TITLE
Akka licensing

### DISCUSSION THREAD

**Discussion by skipper1982** [ScalaConsultants/mesmer#512] [General]
As most of us know, Lightbend is changing the license for Akka to [BSL](https://www.lightbend.com/akka/license). I've digged into their [FAQ page](https://www.lightbend.com/akka/license-faq), and here are my direct conclusions (I would be grateful, if someone can take another look and make sure I haven't missed or misread something):

- Only future versions of Akka modules would be under BSL, so we can safely use and support the current versions
- Future modules versions will fallback to Apache license after 3 years since release
- Current versions of modules will still be supported (critical bug fixes and security updates) until September 2023
- The BSL license requires a commercial license only to production use. 
- There is something like "Additional Use Grant" that allows Open Source Software to be supported by Lightbend, but that requires email communication with them at akka-license@lightbend.com. The example of this working is Play Framework.

Here are some indirect conclusions that may require further verifying, maybe in direct communication with Lightbend:
- Since Mesmer is a library, it is not really a production use, so I would think Mesmer it's not a subject for the commercial license
- If it is a subject for the commercial license after all, it still can be free if the annual revenue of the company doesn't exceed $25m

Also, there are some things, that I'm not really clear about, or require further consideration:

1.  Is Mesmer Open-Source Software? If it is, we can get Lightbend support for akka versions in a similar way Play Framework does.
2. If after all, we do need a commercial license for future versions of Akka modules, and we don't decide to pursue it, what path do we choose? (do we still support all the versions of akka that are under Apache license, including the ones that are falling back to Apache license after 3 years since release? Or do we just sunset the Akka support?)
3. Should we pursue the direct communication with Lightbend to explore further?

@jczuchnowski @adamgadomski @lgajowy Could you take a look and comment on ☝️ ?

**Comment by adamgadomski** [comment]
@skipper1982  thank you for the research and conclusions.

ad.1) So as long as Mesmer is an open-source project, we can use Akka versions without buying a license, right?

ad.2) I think that regardless we need the Akka license we should continue to support Akka libraries, but ofc let's discuss it further - a big amount of work has been done to achieve the current state

ad.3) it will be a good approach - to be on the same page and to have a clear view of using the Akka license

4.) I think that we also should consider instrumenting the Pekko project since it's a fork of the Akka project just before changing the licensing approach
https://cwiki.apache.org/confluence/display/INCUBATOR/PekkoProposal?desktop=true&macroName=markdown

5.) Annual revenue threshold at $25m gives us feedback that still some growing companies would be able to use Akka without buying a new license, so also Mesmer would be a good fit for them

@jczuchnowski @lgajowy please share your thoughts

**Comment by lgajowy** [comment]
> Since Mesmer is a library, it is not really a production use, so I would think Mesmer it's not a subject for the commercial license

[Similar take from a Kamon maintainer](https://github.com/kamon-io/Kamon/issues/1207#issuecomment-1248227759). Kamon is a similar product to Mesmer, especially in the sense of "not really using" Akka.   

I haven't found any thread in open telemetry repositories that would be about the license change. I wonder what is their stance?

> Is Mesmer Open-Source Software? If it is, we can get Lightbend support for Akka versions in a similar way Play Framework does.

[Yes, Apache 2.0 licensed](https://github.com/ScalaConsultants/mesmer/blob/main/LICENSE). I'm still not sure if we need the grant (I'm educating myself). Even if the worst case will happen, i.e. such a grant is needed and we don't get it, I'm ok to drop Akka support and focus on other libraries.

**Comment by lgajowy** [comment]
Started a thread in [opentelemetry-java-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/7097) repo
